src/app/panels/image-panel/image-panel.component.ts

```diff
@@ -1,13 +1,93 @@
-import { Component } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { EVTModelService } from 'src/app/services/evt-model.service';
 import { AppConfig } from '../../app.config';
 
 @Component({
   selector: 'evt-image-panel',
   templateUrl: './image-panel.component.html',
   styleUrls: ['./image-panel.component.scss'],
 })
-export class ImagePanelComponent {
+export class ImagePanelComponent implements OnInit  {
   manifest = AppConfig.evtSettings.files.manifestURL !== '' && !!AppConfig.evtSettings.files.manifestURL
     ? AppConfig.evtSettings.files.manifestURL
     : undefined;
+
+  get imageIndex() { return this._imageIndex; }
+
+  constructor(
+    public evtModelService: EVTModelService,
+    public http: HttpClient,
+    private cdref: ChangeDetectorRef,
+  ) {
+  }
+  public msDesc$ = this.evtModelService.msDesc$;
+  public secondaryContent = '';
+  private showSecondaryContent = false;
+  public msDescOpen = false;
+  public selectedPage;
+  public msDescID = '';
+  public value: number;
+  public images = [];
+  public currentImg: string;
+  // tslint:disable-next-line: variable-name
+  private _imageIndex: number;
+  public page = 0;
+
+  toggleImg(event){
+    this._imageIndex = this.images.indexOf(event);
+  }
+
+  eventHandler(event: number){
+    this.page = event;
+    this.currentImg = this.images[this.page].label;
+    this.cdref.detectChanges();
+  }
+
+  ngOnInit() {
+    // tslint:disable-next-line: no-any
+    this.http.get(`${this.manifest}`).subscribe((item: any) => {
+      item.sequences.forEach((value) => {
+        this.images.push(...value.canvases);
+      });
+      this._imageIndex = 0;
+      this.currentImg = this.images[this.page].label;
+      this.cdref.detectChanges();
```

Comment:
Why do you call explicitly change detection?

Reply:
I added change detection because otherwise the error "ERROR Error: NG0100: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checking" would appear in the console, since every time the image name is updated without the page being refreshed

Reply:
Usually such errors is due to an improper initialisation.